### PR TITLE
[RecoveryServices.Backup] Breaking change release

### PR DESCRIPTION
**Pull request by siddharth7**
## Description
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/210

## Checklist

- [X] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [X] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [X] For public API changes to cmdlets:
    - [X] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [X] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by cormacpayne** [comment]
@siddharth7 Hey Siddharth, I re-targeted your PR at the new default branch, `master`, which contains a bulk of the current changes moving from `AzureRM` to `Az`. You will need to pull the latest changes from this branch and resolve the merge conflicts -- the `AzureRM` markdown files have been removed and replaced by the corresponding `Az` files, so you will need to make the corresponding changes there, and the `ChangeLog.md` file has been cleared, so you will need to make sure the only header remaining is `## Current Release`. Feel free to ping me if you have any questions.

**Comment by siddharth7** [comment]
@azuresdkci Build this please

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Test/ScenarioTests/AzureFiles/ProtectionCheckTests.ps1 [hunk 1]
```diff
@@ -20,6 +20,7 @@ $fileShareName = "AzureFileShare;pstestfs1bca8f8e"
 $saName = "pstestsa1bca8f8e"
 $skuName="Standard_LRS"
 $policyName = "AFSBackupPolicy"
+$storageAccountId = "/subscriptions/da364f0f-307b-41c9-9d47-b7413ec45535/resourceGroups/pstestFSRG1bca8f8e/providers/Microsoft.Storage/storageAccounts/pstestsa1bca8f8e"
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Test/ScenarioTests/AzureFiles/ProtectionCheckTests.ps1]: Please don't use hard-coded storage accounts - you shoudl create any resources required in the test

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup/Cmdlets/Backup/GetAzureRmRecoveryServicesBackupStatus.cs [hunk 2]
```diff
@@ -56,42 +59,61 @@ public class GetAzureRmRecoveryServicesBackupStatus : RecoveryServicesBackupCmdl
...
         [Parameter(ParameterSetName = IdParamSet, ValueFromPipelineByPropertyName = true,
             HelpMessage = ParamHelpMsgs.ProtectionCheck.ResourceId, Mandatory = true)]
+        [Parameter(ParameterSetName = IdWorkloadParamSet, ValueFromPipelineByPropertyName = true,
+            HelpMessage = ParamHelpMsgs.ProtectionCheck.ResourceId, Mandatory = true)]
         public string ResourceId { get; set; }
 
+
+        [Parameter(ParameterSetName = IdWorkloadParamSet, ValueFromPipelineByPropertyName = true,
+          HelpMessage = ParamHelpMsgs.ProtectionCheck.ProtectableObjName, Mandatory = true)]
+        public string ProtectableObjName { get; set; }
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup/Cmdlets/Backup/GetAzureRmRecoveryServicesBackupStatus.cs]: Use ProtectabeObjectName - You can have a parameter alias to shorter names if you like.

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Test/ScenarioTests/AzureFiles/ProtectionCheckTests.ps1 (hunk 1 above)
Comment (siddharth7) [reply src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Test/ScenarioTests/AzureFiles/ProtectionCheckTests.ps1]: There is no support for storage dataplane SDK calls in the test framework due to which all the resources are pre created for AzureFiles workload

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup/Cmdlets/Backup/GetAzureRmRecoveryServicesBackupStatus.cs (hunk 2 above)
Comment (siddharth7) [reply src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup/Cmdlets/Backup/GetAzureRmRecoveryServicesBackupStatus.cs]: Done

**Pull request merged**